spec/datadog/profiling/tag_builder_spec.rb

```diff
@@ -78,5 +78,33 @@
         expect(result).to include('ascii-key' => 'ascii-value')
       end
     end
+
+    describe 'source code integration' do
+      context 'when git environment is available' do
+        before do
+          allow(Datadog::Core::Environment::Git).to receive(:git_repository_url).and_return(
+            'git_repository_url'
+          )
+          allow(Datadog::Core::Environment::Git).to receive(:git_commit_sha).and_return('git_commit_sha')
+        end
+
+        it 'includes the git repository URL and commit SHA' do
+          expect(call).to include(
+            '_dd.git.repository_url' => 'git_repository_url', '_dd.git.commit.sha' => 'git_commit_sha'
+          )
+        end
+      end
+
+      context 'when git environment is not available' do
+        before do
+          allow(Datadog::Core::Environment::Git).to receive(:git_repository_url).and_return(nil)
+          allow(Datadog::Core::Environment::Git).to receive(:git_commit_sha).and_return(nil)
+        end
+
+        it 'includes the git repository URL and commit SHA' do
+          expect(call).to_not include('_dd.git.repository_url', '_dd.git.commit.sha')
+        end
```

Comment:
I suspect this is incorrect...

The `_dd.` prefix I think is only for tracing? Looking at the [source code integration docs for Ruby](https://docs.datadoghq.com/integrations/guide/source-code-integration/?tab=ruby) the correct names should be `git.repository_url` and `git.commit.sha`.

I just tried it locally, by quickly doing this

```diff
-        tags[TAG_REPOSITORY_URL] = git_repository_url if git_repository_url
-        tags[TAG_COMMIT_SHA] = git_commit_sha if git_commit_sha
+        tags['git.repository_url'] = git_repository_url if git_repository_url
+        tags['git.commit.sha'] = git_commit_sha if git_commit_sha
```

and it worked for me.

Here's a [quick link to the profiles I got](https://ddstaging.datadoghq.com/profiling/search?query=service%3Aivoanjo-testing%2A%20&env=staging&my_code=disabled&service=ivoanjo-testing-new&version=2&viz=stream&start=1708005474337&end=1708009074337&paused=false):

* `ivoanjo-testing-old` works, sets the `DD_TAGS` manually
* `ivoanjo-testing-new` is this PR, doesn't work
* `ivoanjo-testing-new-no-prefix` is this PR + the change above

![image](https://github.com/DataDog/dd-trace-rb/assets/2785847/4c787a3c-5d12-4a81-9f47-8a33a89031f4)

Reply:
I... think the `_dd.` prefix for tracing is just a "flag" to tell the backend this is internal stuff to now show by default in the UX, but I'm not 100% sure ;)

Reply:
@ivoanjo fixed

Reply:
Thanks, [can confirm it works](https://ddstaging.datadoghq.com/profiling/search?query=service%3Aivoanjo-testing%2A%20&env=staging&event=AgAAAY2xV_V1B-Lw9QAAAAAAAAAYAAAAAEFZMnhWX2FwQUFEVjJvRWVVN3hYblFBQQAAACQAAAAAMDE4ZGIxNTgtMDQyNy00NDdhLWE1Y2EtYmE5OTYwMmYwZDg3&my_code=disabled&p_tl_tb=1708077344531000064&p_tl_tf_f_1=3866099968&p_tl_tf_us_0=0&p_tl_tf_us_1=3866099968&profileId=AY2xV_apAADV2oEeU7xXnQAA&profileViz=flame_graph&service=ivoanjo-testing-new&version=2&viz=stream&start=1708073757483&end=1708077357483&paused=false)!